packages/blade/src/components/StepGroup/StepGroup.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import type { StepGroupContextType, StepGroupProps } from './types';
+import { StepGroupContext, useStepGroup } from './StepGroupContext';
+import { componentIds } from './componentIds';
+import BaseBox from '~components/Box/BaseBox';
+import { getStyledProps } from '~components/Box/styledProps';
+import { getComponentId } from '~utils/isValidAllowedChildren';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+
+const useChildrenWithIndexes = ({
```

Comment:
Why do we need to precalculate this? This could cause issues and break composition, imagine someone created a reusable StepItem component and used that... 


```js
const StepItemWithReactRouter = (props) => {
  const { to, ...rest } = useLinkHandler(props);
  return <StepItem onClick={to} {...rest} />
}
```

Can we not do something similar to ListItem via React Context?

Reply:
In ListItem also we're doing this only 
<img width="1123" alt="image" src="https://github.com/razorpay/blade/assets/30949385/809b4032-1aa3-425b-90b8-8ad95347e117">

We need to know the index of the item and this seems to be the only way 🤔 

With context we can only get the outer information but not know item's position right?

Reply:
And nope. StepItemLink kind of thing won't be possible. I think that is the case with our other components as well

Reply:
> And nope. StepItemLink kind of thing won't be possible

We should add a storybook example on how StepItem will work with [ReactRouter](https://blade.razorpay.com/?path=/docs/guides-integrations-routing-libraries--docs) then.

Reply:
already added 👯